installer/configurators/common.py

```diff
@@ -0,0 +1,130 @@
+
+
+def get_terraform_folder():
+    return get_installer_root() + "/installer/terraform/"
+
+
+def get_tfvars_file():
+    return get_terraform_folder() + "terraform.tfvars"
+
+
+def passwd_generator():
+    """
+        Random password generator for jazz_ui admin email ID login
+    """
+    alphabet = string.ascii_letters + string.digits
+    return ''.join(secrets.choice(alphabet) for i in range(8))
+
+
+def get_account_id():
+    return boto3.client('sts').get_caller_identity().get('Account')
+
+
+def update_main_terraform_vars(branch, adminemail, stackprefix, region, tags):
+    """
+        Method parse the parameters send from run.py and these common variables
+        are set in terraform.tfvars and other files needed
+    """
+    jazz_branch = branch
+    cognito_details = [adminemail, passwd_generator()]
+    jazz_account_id = get_account_id()
+    prefix = stackprefix.lower()
+    tag_exempt = (datetime.datetime.today() +
+                  datetime.timedelta(days=1)).strftime("%m/%d/%Y")
+
+    # ----------------------------------------------------------
+    # Populate Terraform variables in terraform.tfvars
+    # -----------------------------------------------------------
+
+    # populating BRANCH name
+    replace_tfvars('github_branch', jazz_branch, get_tfvars_file())
+
+    # populating region
+    replace_tfvars('region', region, get_tfvars_file())
+    os.environ['AWS_DEFAULT_REGION'] = region
+
+    # Populating Jazz Account ID
+    replace_tfvars('jazz_accountid', jazz_account_id, get_tfvars_file())
+
+    # Populating Cognito Details
+    replace_tfvars('cognito_pool_username', cognito_details[0],
+                   get_tfvars_file())
+    replace_tfvars('cognito_pool_password', cognito_details[1],
+                   get_tfvars_file())
+
+    # Generate ACL DB password
+    replace_tfvars('acl_db_password', uuid.uuid4().hex,
+                   get_tfvars_file())
+
+    # Set the default values of scm_username/passwd to the cognito credentials
+    # If using existing SCM instance, that scenario (e.g. 1) will obviously overwrite these values.
+
+    replace_tfvars('scm_username', cognito_details[0], get_tfvars_file())
+    replace_tfvars('scm_passwd', cognito_details[1], get_tfvars_file())
+
+    # Populating Jazz Tag env
+    replace_tfvars('envPrefix', prefix, get_tfvars_file())
+    replace_tfvars('tagsExempt', tag_exempt, get_tfvars_file())
+
+    # Populating user tags
+    if tags:
+        input_tags = prepare_tags(tags)
+        try:
+            aws_tags, aws_formatted_tags = validate_replication_tags(input_tags)
+            replace_tfvars("aws_tags", str(aws_tags), get_tfvars_file())
+            replace_tfvars("additional_tags", str(aws_tags), get_tfvars_file(), quoteVal=False)
+        except ValueError as err:
+            print("Invalid Tag!" + str(err))
+            exit(0)
+
+
+# Uses sed to modify the values of key-value pairs within a file
```

Comment:
Comment is no longer applicable. Also the old code below can go away too.

Reply:
Yep, thx.